javaparser-core/src/main/java/com/github/javaparser/resolution/types/ResolvedReferenceType.java

```diff
@@ -294,6 +298,7 @@ public ResolvedTypeParametersMap typeParametersMap() {
 
     /**
      * Corresponding TypeDeclaration
+     * TODO: Convert to Optional
```

Comment:
I plan to do this TODO within this PR so that users have one change to optional to do, as opposed to several smaller changes. 

If anyone has any problems with such a big change, I am more than happy to work with library users to upgrade! :smile:

Reply:
done